### ISSUE TITLE
Compiler plugin is explicit and cannot opt-out

### ISSUE THREAD

**Issue by ppslim** [rjaros/kvision#142]
Although I have been able to establish the answer for myself, it did take me looking at the plugin source just to get an answer.

The documentation talks up the compile plugin for producing common and frontend boilerplate code. However, other than a statement of "if you are using the compiler plugin", there was no detail indicating how I could control if boilerplate code applies or not.

The simpler answer is, it is all or nothing. You load it and the compiler is responsible, or you don't load it and you are responsible. There is no middle ground, or opt-out or detection of manual definitions capability.

The docs do state it will generate it for you, but do not cover how to enable it, how to disable or and/or what the implications may be other than the common/frontend boilerplate.

In my case, I cannot provide my own Manager and thus, cannot provide custom http method or route names. I am forced to use internal route names and to use the POST method.

Any attempt to do so, simply results in messages of re-declaration and obvious conflict between the two definitions as well.

Overall, this means the following key items would have been of benefit to me.
- The `KVService` annotation to provide opt-out capability 
- or; detection if common already has a manager / frontend has a matching service outside of the generated source directories
- or (preferable); new annotations targeting `AnnotationTarget.FUNCTION` that can be applied to the method defined under the KVService interface, which the compiler plugin can use to infer method, or route or both
- Documentation making clear what is lost by disabling the plugin, over just the inline notes all over part 3

Example function annotations for the interface methods
```
@Target(AnnotationTarget.FUNCTION)
annotation class KVServiceBinding(val method: String, val route: String)
@Target(AnnotationTarget.FUNCTION)
annotation class KVServiceBindingMethod(val method: String)
@Target(AnnotationTarget.FUNCTION)
annotation class KVServiceBindingRoute(val route: String)
```

**Comment by rjaros** [comment]
Function annotations is an interesting idea. I'll try to implement this.

**Comment by rjaros** [comment]
Implemented in 3.7.0
